dvc/repo/__init__.py

```diff
@@ -457,8 +457,13 @@ def open(self, path, remote=None, mode="r", encoding=None):
         try:
             with self._open(path, remote, mode, encoding) as fd:
                 yield fd
-        except FileNotFoundError:
-            raise OutputFileMissingError(relpath(path, self.root_dir))
+        except (FileNotFoundError, OutputNotFoundError):
+            try:
+                with self.tree.open(path, mode, encoding) as fd:
+                    yield fd
+
+            except FileNotFoundError:
+                raise FileMissingError(relpath(path, self.root_dir))
 
     def _open(self, path, remote=None, mode="r", encoding=None):
         out, = self.find_outs_by_path(path)
```

Comment:
Btw, this one doesn't handle `cache=False` cases, as it will try looking for uncached file in the remote and will raise an error. CC @Suor

Reply:
It should be done now.